### PR TITLE
#24 Implement watchDynamicTranslatedValue service

### LINKED ISSUE #24: Implement 'watchDynamicTranslatedValue' factory
Since there are a lot of places we need to use 'watchDynamicTranslatedValue', implement it inside common-libs would be more reasonable.

Details [here](https://cryptpad.fr/pad/#/2/pad/edit/b8GZMSd-UgYdNqV2rQ931G7C/) : [i18n Translation] %s is the result of a function

### PR DESCRIPTION
**Pull request by ppppbn**

### CODE REVIEW

File: src/frontend/js/modules/i18n/i18n.service.js [hunk 1]
```diff
@@ -43,5 +43,22 @@ require('./i18n-string.service.js');
       function isI18nString(text) {
         return text instanceof EsnI18nString;
       }
+    })
+    /**
+       * When the value of a dynamic translated text (%s) is relied on the result of a function
```
Comment (tuanlc) [inline src/frontend/js/modules/i18n/i18n.service.js]: wrong indent

**Review by tuanlc — DISMISSED** [review]

File: src/frontend/js/modules/i18n/i18n.service.js (hunk 1 above)
Comment (ppppbn) [reply src/frontend/js/modules/i18n/i18n.service.js]: Re-indent this. Thanks!

**Review by tuanlc — DISMISSED** [review]

File: src/frontend/js/modules/i18n/i18n.service.js [hunk 2]
```diff
@@ -43,5 +43,22 @@ require('./i18n-string.service.js');
       function isI18nString(text) {
         return text instanceof EsnI18nString;
       }
+    })
```
Comment (piavgh) [inline src/frontend/js/modules/i18n/i18n.service.js:46]: We should have a line break here

File: src/frontend/js/modules/i18n/i18n.service.js [hunk 3]
```diff
@@ -43,5 +43,22 @@ require('./i18n-string.service.js');
...
+    /**
+     * When the value of a dynamic translated text (%s) is relied on the result of a function
+     * We'll watch the translated text when it changed by the `get` method of Object.defineProperty
+     * The `get` method should return the updated value
+     *
+     * @param {Object} object         Base object to add property
+     * @param {Object} property       Name of property to watch
+     * @param {Function} callback     Function ran every time we get property value
+    */
+
```
Comment (piavgh) [inline src/frontend/js/modules/i18n/i18n.service.js]: We should NOT have a line break here

**Review by piavgh — DISMISSED** [review]

**Review by MichaelBailly — APPROVED** [review]

**Review by tuanlc — APPROVED** [review]

**Pull request merged**